python/tk_desktop/desktop_window.py

```diff
@@ -1260,8 +1292,30 @@ def __launch_app_proxy_for_project(self, project, requested_pipeline_configurati
         # From this point on, we don't touch the UI anymore.
         self.project_overlay.start_progress()
 
+        self.project_overlay.report_progress(0.05, "Downloading configuration.")
+        self._current_download_thread = ConfigDownloadThread(
```

Comment:
What UI is available while this is happening in the bg? Anything that could get the user in a weird state? For example, is the context menu available? Just curious.

Reply:
Yes, the context menu is indeed available but only shows the different pipeline configurations accessible at this point. Just like before you can switch pipeline configuration if you want.

I've added a few comments to the code however to clarify the story around threading.